### PR TITLE
Refactor Kafka producer to use setPollInterval

### PR DESCRIPTION
**Pull request by ottomata**

### CODE REVIEW

File: config/config.wikimedia.yaml [hunk 1]
```diff
@@ -125,10 +129,18 @@ services:
             # HastyProducer doesn't block HTTP clients, so we can
             # afford to wait for a largish batch size.
             queue.buffering.max.ms: 1000
+            batch.num.messages: 500
+            # Custom kafka config, will call producer.setPollInterval
+            # with this value if set.
+            producer.poll.interval.ms: 200
         guaranteed:
           conf:
             # GuaranteedProducer doese block HTTP clients, so we
```
Comment (Pchelolo) [inline config/config.wikimedia.yaml]: Typo

File: config/config.wikimedia.yaml [hunk 2]
```diff
@@ -125,10 +129,18 @@ services:
...
             # should use a smaller batch size. Waiting 50 ms
             # will allow for some kafka produce reqeust batching
             # while minimizing latency on the HTTP client.
-            queue.buffering.max.ms: 50
+            queue.buffering.max.ms: 0
```
Comment (Pchelolo) [inline config/config.wikimedia.yaml:140]: This ain't match with the comment anymore. And why 0?

File: lib/kafka.js [hunk 3]
```diff
@@ -201,40 +151,63 @@ class HastyProducer extends kafka.Producer {
...
  * @param {Object} logger bunyan logger.  If given, will be used for rdkafka event.log callback
  * @return {Promise<Object>}
  */
-HastyProducer.factory = (config = {}, topicConfig = {}, logger = null) => {
+function producerFactory(producerType = 'guaranteed', config = {}, topicConfig = {}, logger) {
+    // Map producerType to class.
+    const producerTypes = {
+        guaranteed: GuaranteedProducer,
+        hasty: HastyProducer
+    };
```
Comment (Pchelolo) [inline lib/kafka.js]: We can make this map a module-level constant.

**Comment by Pchelolo** [comment]
Couple of minors inlined, otherwise LGTM.

File: config/config.wikimedia.yaml (hunk 2 above)
Comment (ottomata) [reply config/config.wikimedia.yaml:140]: 0 is actually the default the librdkafka uses.  In my perf tests, this seemed to get the best throughput.  Still remains to be seen what is the best combo in prod.

**Pull request merged**